src/reporters/udp_sender.js

```diff
@@ -57,12 +58,30 @@ export default class UDPSender {
         });
     }
 
+    _calcBatchSize(batch: Batch) {
+        let thriftBatch = this._thrift.getType('Agent::emitBatch_args');
+        let buffer: Buffer = thriftBatch.toBufferResult({'batch': batch}).value;
+        return buffer.length;
+    }
+
     _calcSpanSize(span: any): number {
         let thriftJaegerSpan = this._thrift.getType('Span');
         let buffer: Buffer = thriftJaegerSpan.toBufferResult(span).value;
         return buffer.length;
     }
 
+    setProcess(process: Process): void {
+        // This function is only called once during reporter construction, and thus will
+        // give us the length of the batch before any spans have been added to _spanBuffer.
+        this._process = process;
+        this._batch = {
+            'process': this._process,
+            'spans': this._spanBuffer
+        };
+        this._emitSpanBatchOverhead = this._calcBatchSize(this._batch);
+        this._maxSpanBytes -= this._emitSpanBatchOverhead;
```

Comment:
So this bug was masked by the fact that each of my tests happens to set maxSpanBytes for its own purposes.